runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -308,100 +313,151 @@ end
 
 ---@param self vim.treesitter.highlighter
 ---@param buf integer
----@param line integer
+---@param range_br integer
+---@param range_bc integer
+---@param range_er integer
+---@param range_ec integer
 ---@param on_spell boolean
 ---@param on_conceal boolean
-local function on_line_impl(self, buf, line, on_spell, on_conceal)
-  self._conceal_checked[line] = true
+local function on_range_impl(
+  self,
+  buf,
+  range_br,
+  range_bc,
+  range_er,
+  range_ec,
```

Comment:
Again, prefer the terms `start/end` for ranges. So this would be
```suggestion
  start_row,
  start_col,
  end_row,
  end_col,
```

Reply:
Is `start`/`end` preferred over `begin`/`end`?

Reply:
This conflicts with the node range:
```lua
      local start_row, start_col, end_row, end_col = Range.unpack4(range)
```

what should it be named?

Reply:
> Is `start`/`end` preferred over `begin`/`end`?

yes